http-clients/src/main/java/com/palantir/remoting/http/OptionalAwareContract.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.remoting.http;
+
+import com.google.common.base.Optional;
+import feign.Contract;
+import feign.Feign;
+import feign.MethodMetadata;
+import java.lang.reflect.Method;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class OptionalAwareContract implements Contract {
```

Comment:
also I think for future-proofing here we may want to call this `GuavaOptionalAwareContract`